### PR TITLE
[zephyr] Added platform configuration to enable ICD support

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Changed Zephyr and nrfconnect platforms configuration to move from supporting SED to ICD:
* Removed custom subscriptions handling, to meet the ICD max subscription interval requirement.
* Renamed SED kconfig option names and descriptions to match the ICD related ones.
* Introduced new kconfig options allowing to configure idle mode interval, active mode interval, active mode threshold and number of clients per fabric.

### CODE REVIEW

**Review by markaj-nordic — APPROVED** [review]

File: config/zephyr/Kconfig [hunk 1]
```diff
@@ -323,37 +323,66 @@ config CHIP_ENABLE_DNS_CLIENT
...
+config CHIP_ICD_ACTIVE_MODE_THRESHOLD
+	int "Intermittently Connected Device active mode threshold (ms)"
+	default 300
+	help
+	  Provides the Intermittently Connected Device active mode threshold is milliseconds.
+	  It determines the minimum amount of time the device shall stay in the active mode after the network activity.
+
+config CHIP_ICD_CLIENTS_PER_FABRIC
+	int "Intermittently Connected Device number of clients per fabric"
+	default 2
```
[the remark this answers is not in the document]
Comment (kkasperczyk-no) [reply config/zephyr/Kconfig:376]: Yes, the default values were based on values used by the defines in the Matter core. My intention was to provide only a configuration layer allowing to modify the defaults using Kconfigs. According to the spec, the minimal value for this option should be 1.

**Review by tima-q — APPROVED** [review]

File: config/zephyr/Kconfig [hunk 2]
```diff
@@ -323,37 +323,66 @@ config CHIP_ENABLE_DNS_CLIENT
...
+	  device is in the active mode. It determines the fastest frequency at which the device will be able
+	  to receive the messages in the active mode. The CHIP_ICD_FAST_POLLING_INTERVAL shall be smaller than
+	  CHIP_ICD_ACTIVE_MODE_INTERVAL.
 
-endif # CHIP_ENABLE_SLEEPY_END_DEVICE_SUPPORT
+config CHIP_ICD_IDLE_MODE_INTERVAL
+	int "Intermittently Connected Device idle mode interval (s)"
+	default 120
+	help
+	  Provides the Intermittently Connected Device idle mode interval is seconds.
```
Comment (Damian-Nordic) [inline config/zephyr/Kconfig:356]:
```suggestion
	  Provides the Intermittently Connected Device idle mode interval in seconds.
```

File: config/zephyr/Kconfig [hunk 3]
```diff
@@ -323,37 +323,66 @@ config CHIP_ENABLE_DNS_CLIENT
...
+	  It determines the maximum amount of time the device can stay in the idle mode, which means the
+	  device may be unreachable and not able to receive messages.
+
+config CHIP_ICD_ACTIVE_MODE_INTERVAL
+	int "Intermittently Connected Device active mode interval (ms)"
+	default 300
+	help
+	  Provides the Intermittently Connected Device active mode interval is milliseconds.
```
Comment (Damian-Nordic) [inline config/zephyr/Kconfig:364]:
```suggestion
	  Provides the Intermittently Connected Device active mode interval in milliseconds.
```

File: config/zephyr/Kconfig [hunk 4]
```diff
@@ -323,37 +323,66 @@ config CHIP_ENABLE_DNS_CLIENT
...
+	  It determines the minimum amount of time the device shall stay in the active mode.
+
+config CHIP_ICD_ACTIVE_MODE_THRESHOLD
+	int "Intermittently Connected Device active mode threshold (ms)"
+	default 300
+	help
+	  Provides the Intermittently Connected Device active mode threshold is milliseconds.
```
Comment (Damian-Nordic) [inline config/zephyr/Kconfig:371]:
```suggestion
	  Provides the Intermittently Connected Device active mode threshold in milliseconds.
```

**Review by Damian-Nordic — APPROVED** [review]

**Pull request merged**